### PR TITLE
Extensibility form reorder

### PR DESCRIPTION
**Pull request by qbalukom**
**Description**

Changes proposed in this pull request:

- improve extensibility forms to allow reordering and more structured forms

### CODE REVIEW

**Review by dariadomagala — CHANGES_REQUESTED** [review]
I don't see all values I have in my configuration

What I see
Screen Shot 2022-07-20 at 07 57 38

What's in my config 
```
    {
        "path": "spec.ports[]",
        "simple": true,
        "children": [
          {
            "path": "name",
            "simple": true
          },
          {
            "path": "protocol",
            "simple": true
          },
          {
            "path": "port",
            "simple": true
          },
          {
            "path": "targetPort",
            "simple": true
          }
        ]
      }
```

What's in the resource, so we know this entry exists:
```
spec:
  ports:
    - name: dd3
      protocol: TCP
      port: 5333
      targetPort: 5333
```

File: docs/extensibility/resources.md [hunk 1]
```diff
@@ -40,8 +40,9 @@ If you target elements of an array rather that the array itself, you can use `it
 
 ### Item parameters
 
-- **path** - _[required]_ path to the property that you want to display in the form. In the case of an array, the array index is omitted. For example, if `spec.items` is an array and you want to display `name` for each item, the path is `spec.items.name`.
+- **path** - _[required]_ path to the property that you want to display in the form.
 - **widget** - optional widget used to render the field referred to by the **path** property. If you don't provide the widget, a default handler is used depending on the data type provided in the schema. For more information about the available widgets, see [Form widgets](form-widgets.md).
+- **children** - child widgets used for grouping. Child paths are relative to it's parent.
```
[the remark this answers is not in the document]
Comment (qbalukom) [reply docs/extensibility/resources.md:45]: no, because it's not required

**Comment by wchrapka** [comment]
This spec works on main, but doesn't on this PR:

```
  "form": [
      {
        "path": "spec.aorb"
      },
      { "path": "spec.a" },
      { "path": "spec.b" }
    ],
(...schema:)
  "spec": {
          "description": "LogPipelineSpec defines the desired state of LogPipeline",
          "type": "object",
          "if": {
            "properties": { "aorb": { "const": "a" } }
          },
          "then": {
            "properties": { "a": { "type": "string" } }
          },
          "else": {
            "properties": { "b": { "type": "integer" } }
          },
          "properties": {
            "aorb": {
              "type": "string",
              "enum": ["a", "b"],
              "default": "b"
            },
            "enableUnsupportedPlugins": {
              "type": "boolean"
            },

**Review by dariadomagala — APPROVED** [review]

File: docs/extensibility/resources.md (hunk 1 above)
Comment (grego952) [inline docs/extensibility/resources.md:45]:
```suggestion
- **children** - child widgets used for grouping. Child paths are relative to its parent.
```

**Review by dariadomagala — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**